mutation_reader.cc

```diff
@@ -2052,7 +2052,13 @@ class queue_reader final : public flat_mutation_reader::impl {
         return _full->get_future();
     }
     virtual future<> next_partition() override {
-        return make_exception_future<>(make_backtraced_exception_ptr<std::bad_function_call>());
+        clear_buffer_to_next_partition();
+        if (is_buffer_empty() && !is_end_of_stream()) {
+            return fill_buffer(db::no_timeout).then([this] {
+                return next_partition();
+            });
+        }
+        return make_ready_future<>();
```

Comment:
@denesb please review this

the db::no_timeout seems wrong, but it's a problem in the API itself. Will go away when we remove timeout from fill_buffer.

Just reading stuff from queue_reader until we exhaust the partition is wasteful, but that's a queue_reader limitation, there is limited interaction with the producer. Do we really need queue_reader? I think the virtual reader could be a from-scratch reader and then communicate with execute() to ask it to skip. Anyway this is okay for the first iteration.